source/AI.cpp

```diff
@@ -3514,13 +3514,40 @@ void AI::MovePlayer(Ship &ship, const PlayerInfo &player, Command &activeCommand
 	AimTurrets(ship, firingCommands, !Preferences::Has("Turrets focus fire"));
 	if(Preferences::Has("Automatic firing") && !ship.IsBoarding()
 			&& !(autoPilot | activeCommands).Has(Command::LAND | Command::JUMP | Command::FLEET_JUMP | Command::BOARD)
-			&& (!target || target->GetGovernment()->IsEnemy()))
+			&& (!target || target->GetGovernment()->IsEnemy())
+			&& !isMouseTurningEnabled)
 		AutoFire(ship, firingCommands, false);
+
+	if (activeCommands.Has(Command::MOUSETURNING)) {
+		isMouseTurningEnabled = !isMouseTurningEnabled;
+	}
+
+	if (isMouseTurningEnabled && !ship.IsBoarding() && !ship.IsReversing())
+	{
+		Angle shipAngle = ship.Facing();
+		if (rightMouseButtonHeld)
+		{
+			int index = 0;
+			for(const Hardpoint &hardpoint : ship.Weapons())
+			{
+				if(hardpoint.IsReady() && !hardpoint.GetOutfit()->Icon())
+					firingCommands.SetFire(index);
+				++index;
+			}
```

Comment:
Probably better to set the firing command instead of iterate the hardpoints.  This is a copy of existing code.